### PR TITLE
Improve writef with groups for hash-numbers

### PR DESCRIPTION
**Pull request by mppf**
Allow %{} syntax to escape ### things like %{##.##}
E.g. %{##.##} means the same as ##.##, but the end
of the group is explicit, so you could have another
format right after:
 writef("%{#.#}.%{#.#}", 1.2, 3.4)
would output 1.23.4

### CODE REVIEW

**Comment by mppf** [comment]
@lydia-duncan  can you review this?

**Comment by mppf** [comment]
Passed full local testing.

File: test/io/ferguson/hash_group.good [hunk 1]
```diff
@@ -0,0 +1,10 @@
+  1
+  1.
+  1..
+  1.10
+  1. 2
+  1. 2
+  1  2
```
Comment (lydia-duncan) [inline test/io/ferguson/hash_group.good]: It looks like there's a lot of extra spaces in here, is that intentional?

Comment (mppf) [inline test/io/ferguson/hash_group.good]: Yep. The format is ### - (eg for the first one) meaning it should print out 3 columns at least...

Comment (lydia-duncan) [inline test/io/ferguson/hash_group.good]: Ah, that makes sense.  Looks great!

**Pull request merged**